helix-term/src/commands.rs

```diff
@@ -3896,6 +3901,108 @@ fn surround_delete(cx: &mut Context) {
     })
 }
 
+#[derive(Eq, PartialEq)]
+enum ShellBehavior {
+    Replace,
+    Insert,
+    Append,
+    Filter,
+    None,
+}
+
+fn shell_pipe(cx: &mut Context) {
+    shell(cx, "pipe:", true, ShellBehavior::Replace);
+}
+
+fn shell_pipe_ignore(cx: &mut Context) {
+    shell(cx, "pipe:", true, ShellBehavior::None);
+}
+
+fn shell_insert(cx: &mut Context) {
+    shell(cx, "command:", false, ShellBehavior::Insert);
+}
+
+fn shell_append(cx: &mut Context) {
+    shell(cx, "command:", false, ShellBehavior::Append);
+}
+
+fn shell_filter(cx: &mut Context) {
+    shell(cx, "filter:", true, ShellBehavior::Filter);
+}
+
+fn shell(cx: &mut Context, prompt: &str, pipe: bool, behavior: ShellBehavior) {
+    use std::io::Write;
+    use std::process::{Command, Stdio};
+    let shell = Option::<Vec<_>>::None; // this should come from config, but can't currently
+    let shell = match shell {
+        Some(v) if !v.is_empty() => v,
+        _ => {
+            if cfg!(windows) {
+                vec!["cmd".to_owned(), "/C".to_owned()]
+            } else {
+                vec![
+                    std::env::var("SHELL").unwrap_or_else(|_| "sh".to_owned()),
```

Comment:
I don't think using `SHELL` is a good idea since some shells like fish is not POSIX-compatible even though being set as SHELL at runtime. Maybe we can just use `/bin/sh`?

Reply:
I don't think it's a problem since the user is supposed to execute commands, not helix. Hence we would not have to assume anything about the shell itself.

Reply:
Well, trying to use `-c` with the shell to have it execute a command is assuming something about the shell itself.